FlowCrypt/Functionality/Services/EKMService.swift

```diff
@@ -74,10 +70,14 @@ final class EKMService: EKMServiceType {
         }
     }
 
-    private func fetchCurrentPassphrase() throws -> String? {
+    private func getPassphrase(in viewController: UIViewController) async throws -> String? {
         // If this is called when starting the app, then it doesn't make much difference
         // but conceptually it would be better to look pass phrase both in memory and storage
-        return try appContext.passPhraseService.getPassPhrases(for: appContext.user.email).first(where: { $0.value.isNotEmpty })?.value
+        var passphrase = try appContext.passPhraseService.getPassPhrases(for: appContext.user.email).first(where: { $0.value.isNotEmpty })?.value
+        if passphrase == nil {
+            passphrase = try await requestPassPhraseWithModal(in: viewController)
+        }
+        return passphrase
     }
```

Comment:
could be a bit simpler with 

```swift
if let passPhrase = try appContext.passPhraseService.getPassPhrases(for: appContext.user.email).first(where: { $0.value.isNotEmpty })?.value {
  return passPhrase
}
return try await requestPassPhraseWithModal(in: viewController)
```

`var` should generally be avoided in favor of `let`, everything else being equal.

Reply:
this wasn't addressed, unresolving